app/lib/core/services/service_locator.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:app/core/view_models/nav_view_model.dart';
+import 'package:app/core/view_models/signin_view_model.dart';
+import 'package:app/core/view_models/signout_view_model.dart';
+import 'package:app/core/view_models/welcome_view_model.dart';
+import 'package:get_it/get_it.dart';
+import 'authentication/auth_service.dart';
+import 'navigation/nav_service.dart';
+
+final locator = GetIt.instance;
+
+void setUpLocator() {
+  locator.registerLazySingleton<AuthenticationService>(
+      () => AuthenticationService());
+  locator.registerLazySingleton<NavigationService>(() => NavigationService());
+
+  locator.registerFactory<WelcomeViewModel>(() => WelcomeViewModel());
```

Comment:
Hm, is this the intended use of direct injection for the models? This strikes me as excessive, but I see the usefulness of it as we should technically be able to launch widgets any where we need.

Reply:
We could just have factories in the widgets which I believe is the intended use (like what `loading_overlay_screen.dart` had). They are static and available anywhere.

Reply:
Yes, every viewmodels and services are required to be added in the locator. Without this our app wont be able to communicate data to UI. As for the type, I can remove that. Not really necessary, this is similar to <Widgets> child:

Reply:
That seems anti-pattern to me but OK.